libpromises/matching.c

```diff
@@ -292,7 +292,7 @@ void EscapeSpecialChars(const char *str, char *strEsc, int strEscSz, char *noEsc
                 break;
             }
 
-            strcat(strEsc, noEscSeq);
+            strlcat(strEsc, noEscSeq, strEscSz);
```

Comment:
I think so, so strcat() might be safe here. I will leave strlcat() though unless you are sure and feel strongly about this.